### PR TITLE
Implementing client for User service

### PR DESCRIPTION
**Pull request by ilisau**
In this PR I decompose application and implement client for UserService

### CODE REVIEW

File: src/main/java/com/solvd/qaprotours/service/UserService.java [hunk 1]
```diff
@@ -1,27 +1,56 @@
...
+import feign.codec.ErrorDecoder;
+import org.springframework.cloud.openfeign.FeignClient;
+import org.springframework.stereotype.Component;
+import org.springframework.web.bind.annotation.*;
 
 /**
  * @author Ermakovich Kseniya, Lisov Ilya
  */
+@FeignClient(value = "user-service", path = "/api/v1/users")
 public interface UserService {
```
Comment (h1alexbel) [inline src/main/java/com/solvd/qaprotours/service/UserService.java:19]: @ilisau 
a better naming is ```UserClient```

**Review by h1alexbel — COMMENTED** [review]
@ilisau explain these changes, please. Or name pr properly

File: src/main/java/com/solvd/qaprotours/domain/exception/MicroserviceException.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+import lombok.Getter;
+
+import java.util.Map;
+
+/**
+ * @author Lisov Ilya
+ */
+@Getter
+public class MicroserviceException extends RuntimeException {
```
[the remark this answers is not in the document]
Comment (ilisau) [reply src/main/java/com/solvd/qaprotours/domain/exception/MicroserviceException.java:11]: ErrorDto from UserService (from its ControllerAdvice). I throw it to use ClientErrorDecoder

[the remark this answers is not in the document]
Comment (h1alexbel) [reply src/main/java/com/solvd/qaprotours/domain/exception/MicroserviceException.java:11]: rename it. should be ```ExernalServiceException``` or ```UserServiceException```

Comment (h1alexbel) [inline src/main/java/com/solvd/qaprotours/domain/exception/MicroserviceException.java:11]: rename it. Make it more specific.

**Review by h1alexbel — CHANGES_REQUESTED** [review]

**Review by h1alexbel — APPROVED** [review]

**Pull request merged**